fairlearn/metrics/_metric_frame.py

```diff
@@ -364,6 +392,138 @@ def __init__(
 
+        # Calculate uncertainty intervals using bootstrapped versions of data
+        def _bootstrap_metrics_calc(data):
+            overall_frame = self._build_overall_frame(
+                data, annotated_funcs, cf_list, self._cf_names
+            )
+            by_group_frame = self._build_by_group_frame(
+                data, annotated_funcs, grouping_features
+            )
+            group_min_frame = self.__group(by_group_frame, "min", errors="coerce")
+            group_max_frame = self.__group(by_group_frame, "max", errors="coerce")
+
+            overall_calc_frame, by_group_calc_frame = overall_frame, by_group_frame
+            if (
+                self._user_supplied_callable
+            ):  # This logic is borrowed from .overall/.by_group -- refactor?
+                by_group_calc_frame = by_group_frame.iloc[:, 0]
+                if self.control_levels:
+                    overall_calc_frame = overall_frame.iloc[:, 0]
+                else:
+                    overall_calc_frame = overall_frame.iloc[0]
+
+            diff_overall_frame = self.__difference(
+                by_group_calc_frame, overall_calc_frame
+            )
+            diff_group_frame = self.__difference(by_group_calc_frame, group_min_frame)
+            ratio_overall_frame = self._calc_ratio_overall(
+                by_group_calc_frame, overall_calc_frame
+            )
+            ratio_group_frame = group_min_frame / group_max_frame
+
+            return (
+                overall_frame,
+                by_group_frame,
+                group_min_frame,
+                group_max_frame,
+                diff_group_frame,
+                diff_overall_frame,
+                ratio_group_frame,
+                ratio_overall_frame,
+            )
+
+        self._overall_ci, self._by_group_ci = None, None
+        self.difference_overall_ci, self.difference_group_ci = None, None
+        self.ratio_overall_ci, self.ratio_groups_ci = None, None
+        self.group_min_ci, self.group_max_ci = None, None
+
+        if n_boot is not None:
+            # Calculate all outputs across entire data sample to assist bootstrap bias calculations
+            # NOTE: Consider caching these results for on-demand regular calls?
+            try:  # TODO: Find a better way of checking for functions with non-scalar return types.
+                group_min_output = self.__group(self._by_group, "min")
+                group_max_output = self.__group(self._by_group, "max")
+                difference_group_output = self.__difference(
+                    self.by_group, group_min_output
+                )
+                difference_overall_output = self.__difference(
+                    self.by_group, self.overall
+                )
+                ratio_group_output = group_min_output / group_max_output
+                ratio_overall_output = self._calc_ratio_overall(
+                    self.by_group, self.overall
+                )
+            except:
```

Comment:
Having been bitten by this when bring the PR over, I think it would be best not to completely throw out the actual exception. Given that the raised `ValueError` will likely be the most common for users, I can see the sense in keeping it, but I think we should nest the actual exception inside it.

Reply:
Fixed this mone.